pkg/codegen/schema/schema.go

```diff
@@ -378,6 +380,64 @@ type Resource struct {
 	Methods []*Method
 }
 
+// The set of resource paths where ReplaceOnChanges is true.
+//
+// For example, if you have the following resource struct:
+//
+// Resource A {
+// Properties: {
+// 	 Resource B {
+// 	   Object D: {
+// 	     ReplaceOnChanges: true
+// 	     }
+// 	   Object F: {}
+//     }
+// 	 Object C {
+// 	   ReplaceOnChanges: true
+// 	   }
+//   }
+// }
+//
+// A.ReplaceOnChanges() == [[B, D], [C]]
+func (r *Resource) ReplaceOnChanges() (replaceOnChanges [][]*Property) {
+	for _, p := range r.Properties {
+		if p.ReplaceOnChanges {
+			replaceOnChanges = append(replaceOnChanges, []*Property{p})
+		} else if child, ok := p.Type.(*ResourceType); ok {
```

Comment:
I don't think this is quite right. If a resource has a property that's typed as a `*ResourceType`, I don't think we want to drill into its properties. @lblackstone, correct me if I am wrong.

I think we do want to drill into properties typed as `*ObjectType` (looking at its `.Properties` deeply).

And, in addition to handling `*OptionalType`, I suspect we also need to handle `*UnionType`.

@lblackstone, what about `*MapType` and `*ArrayType`? Is there a syntax that `replaceOnChanges` supports for those?

Reply:
> @lblackstone, what about `*MapType` and `*ArrayType`? Is there a syntax that `replaceOnChanges` supports for those?

Yes, those can be specified like this:
- Map: `['foo.bar', 'baz.*']`
- Array: `['foo[0]', 'bar[*]']`

Reply:
@lblackstone, Is the idea that you can do something like this:
```json
"bar": {
  "type": "array",
  "items": {
    "type": "number",
    "replaceOnChanges": true,
  }
}
```
to generate `'bar[*]'`. 
```json
"bar": {
  "type": "array",
  "replaceOnChanges": true,
  "items": {
    "type": "number"
  }
}
```
already generates `'bar'`. 
And the same thing for maps?

Reply:
Your example looks correct to me.

Also note that specifying the parent implicitly causes a change if any of the dependents change. For example,

`['']` or `['*']` will always cause a replacement since it specifies the root of the resource.

Reply:
This should be resolved now.